modules/tinymce/src/plugins/template/test/ts/browser/TemplateSanityTest.ts

```diff
@@ -76,64 +77,77 @@ describe('browser.tinymce.plugins.template.TemplateSanityTest', () => {
-          UiFinder.notExists(iframeBody, notExistsSelector);
-        }
-      );
+  context('Previewing unparsed content', () => {
+    const unparsedHtml = '<img src="error" onerror="throw new Error();">';
+    const unparsedPreviewHtmlSelector = 'p > img[src="error"][onerror="throw new Error();"]';
+    const parsedPreviewHtmlSelector = 'p > img[src="error"][data-mce-src="error"]';
+
+    const pPreviewAndAssertNoUnparsedContent = async (editor: Editor): Promise<void> => {
+      const assertNoUnparsedContent = (dialogEl: SugarElement<Node>): void => {
+        UiFinder.findIn<HTMLIFrameElement>(dialogEl, 'iframe').fold(
+          () => assert.fail('Preview iframe not found'),
+          (iframe) => {
+            const iframeDoc = iframe.dom.contentDocument;
+            const iframeBody = SugarElement.fromDom(iframeDoc?.body as Node);
+            UiFinder.exists(iframeBody, parsedPreviewHtmlSelector);
+            UiFinder.notExists(iframeBody, unparsedPreviewHtmlSelector);
+          }
+        );
+      };
+
+      try {
+        await pPreviewTemplate(editor, assertNoUnparsedContent);
+      } catch {
+        assert.fail('Unparsed html read');
+      }
     };
 
-    try {
-      await pPreviewTemplate(editor, (dialogEl) => assertPreviewContent(dialogEl, sanitizedPreviewHtmlSelector, unsanitizedPreviewHtmlSelector));
-    } catch {
-      assert.fail('Unsanitized html read');
-    }
+    it('TINY-9244: Parsed html should be shown when previewing template', async () => {
+      const editor = hook.editor();
+      addSettings({
+        templates: [{ title: 'a', description: 'b', content: unparsedHtml }],
+      });
+      await pPreviewAndAssertNoUnparsedContent(editor);
+    });
+
+    it('TINY-9867: Parsed html should be shown when previewing template containing <html> tags', async () => {
+      const editor = hook.editor();
+      addSettings({
+        templates: [{ title: 'a', description: 'b', content: `<html>${unparsedHtml}</html>` }],
+      });
+      await pPreviewAndAssertNoUnparsedContent(editor);
+    });
   });
 
-  const assertFnDoesNotReadUnsanitizedHtmlInDom = async (editor: Editor, fn: (unsanitizedHtml: string) => void | Promise<void>): Promise<void> => {
-    const fnReadsUnsanitizedHtmlInDom = async () => {
-      const unsanitizedHtml = '<img src="error" onerror="window.document.unsanitizedHtmlFn();">';
-      let isUnsanitizedHtmlRead = false;
-      (editor.getDoc() as any).unsanitizedHtmlFn = () => {
-        isUnsanitizedHtmlRead = true;
+  context('Inserting unparsed content', () => {
+    const unparsedHtml = '<img src="error" onerror="window.document.unparsedHtmlFn();">';
+    const assertFnDoesNotReadUnParsedHtmlInDom = async (editor: Editor, fn: (unparsedHtml: string) => void | Promise<void>): Promise<void> => {
+      const fnReadsUnparsedHtmlInDom = async () => {
+        const isUnParsedHtmlRead = Cell(false);
+        (editor.getDoc() as any).unparsedHtmlFn = () => {
```

Comment:
To be safe, it could be removing the `unparsedHtmlFn` from the document, by setting it to null. Maybe in an `after` function within the context?

Reply:
That's a good catch, added.